src/linear_algebra/finite_dimensional.lean

```diff
@@ -828,16 +828,38 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma eq_zero_of_findim_eq_zero [finite_dimensional K V]
+  (hV : findim K V = 0) (x : V) : x = 0 :=
+begin
+  obtain ⟨B, hB⟩ := exists_is_basis_finset K V,
+  have := (findim_eq_card_finset_basis hB).symm,
+  rw [hV, finset.card_eq_zero, finset.eq_empty_iff_forall_not_mem] at this,
+  rw ← is_basis.equiv_fun_total hB x,
+  convert finset.sum_empty,
+  rw finset.univ_eq_empty,
+  rintro ⟨b, hB⟩,
+  exact this b hB,
+end
```

Comment:
Can you instead prove `findim K V = 0 ↔ subsingleton V`?

Reply:
I proved `findim K V = 0 ↔ subsingleton V` using `eq_zero_of_findim_eq_zero` since thats the first proof that came into my head. I'm not sure if there is an easier proof